### DISCUSSION TITLE
Any one see this error after click "live"?

### DISCUSSION THREAD

**Discussion by patatoo** [hacksider/Deep-Live-Cam#482] [Q&A]
latent = source_face.normed_embedding.reshape((1,-1))
AttributeError: 'NoneType' object has no attribute 'normed_embedding'

**Comment by kf0916** [comment]
+1

**Comment by barrywass** [comment]
This is what GPT says

The error message in your screenshot indicates that the source_face object is None. Specifically, it says:

csharp
Copy code
AttributeError: 'NoneType' object has no attribute 'normed_embedding'
This means that source_face, which is expected to be an object with an attribute called normed_embedding, is instead None. In Python, a NoneType object means that the variable has not been assigned any valid object, and thus it cannot have attributes or methods.

Possible causes for this issue:
Incorrect initialization of source_face: The object source_face might not have been properly created or assigned before you try to access its normed_embedding attribute.
Failed operation: Whatever function or process that was supposed to create or retrieve the source_face object could have failed, returning None instead of a valid object.
Data loading issue: If source_face is supposed to represent some loaded data (like a face embedding), the loading process may have failed, resulting in None.
To resolve this issue:
Check the creation or assignment of source_face: Ensure that source_face is being properly initialized and assigned a valid object before you try to access its attributes.

Add checks: Before accessing normed_embedding, you can add a conditional statement to check if source_face is None:

python
Copy code
if source_face is not None:
    latent = source_face.normed_embedding.reshape((1, -1))
else:
    print("source_face is None")
Trace back the function calls: Investigate the function that is supposed to assign or retrieve source_face and ensure it returns a valid object.

By troubleshooting how source_face is being initialized and tracing the data flow, you should be able to fix this issue.

**Reply by milo-jo** [reply]
it worked for you?

**Comment by barrywass** [comment]
Yes it's working really well but I am using a different variant of the program with more options
    On Wednesday, December 11, 2024 at 10:12:22 AM CST, milo-jo ***@***.***> wrote:  
 
 


it worked for you?
